justspacesapp/urls.py

```diff
@@ -13,9 +13,98 @@
     1. Import the include() function: from django.urls import include, path
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
-from django.conf.urls import url, include
-from justspacesapp.views import CreateNewSurvey
+from django.conf.urls import url
+from django.utils.translation import ugettext_lazy as _
+from justspacesapp.views import AgencyCreate, LocationCreate, StudyCreate, \
+                                SurveyListEdit, SurveyListView
+import fobi.views
 
 urlpatterns = [
-    url(r'^$', CreateNewSurvey.as_view(), name='create-new-survey'),
+    url(r'^$',
+        SurveyListView.as_view(),
+        name='home'),
+
+    url(r'agencies/create$',
+        AgencyCreate.as_view(),
+        name='agencies-create'),
+
+    url(r'locations/create$',
+        LocationCreate.as_view(),
+        name='locations-create'),
+
+    url(r'studies/create$',
+        StudyCreate.as_view(),
+        name='studies-create'),
+
+    # Create new survey
+    url(r'surveys/create$',
+        view=fobi.views.create_form_entry,
+        name='fobi.create_form_entry'),
+
+    # Create survey element
+    url(_(r'^surveys/elements/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_element_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_element_entry,
+        name='fobi.add_form_element_entry'),
+
+    # Edit survey element
+    url(_(r'^surveys/elements/edit/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_element_entry,
+        name='fobi.edit_form_element_entry'),
+
+    # Delete survey element
+    url(_(r'^surveys/elements/delete/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_element_entry,
+        name='fobi.delete_form_element_entry'),
+
+    # Create survey handler
+    url(_(r'^surveys/handlers/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_handler_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_handler_entry,
+        name='fobi.add_form_handler_entry'),
+
+    # Edit survey handler
+    url(_(r'^surveys/handlers/edit/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_handler_entry,
+        name='fobi.edit_form_handler_entry'),
+
+    # Delete survey handler
+    url(_(r'^surveys/handlers/delete/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_handler_entry,
+        name='fobi.delete_form_handler_entry'),
+
+    # Edit survey
+    url(_(r'^surveys/edit/(?P<form_entry_id>\d+)/$'),
```

Comment:
More typically, the action comes after the id. (I think Fobi does it the other way around....and I may have done the same in my original PR.) 

But even more typically, the action does not appear in the URL, but rather in the request....[the first comment in this StackOverflow article nicely summarizes the difference.](https://stackoverflow.com/questions/256349/what-are-the-best-common-restful-url-verbs-and-actions) It looks like Fobi includes the action verb in their (function-based?) views, which might make standardizing these URLs a little tricky. 

Can you add a doc string flagging this? (We might want to revisit, as the tool develops.)

Reply:
Fobi does include actions in their function-based views and urls, and these are ported pretty directly. I'll leave a comment for now!

Reply:
Yes, that's a pitfall of function-based views. We can talk about if it's worth fighting against.